### ISSUE TITLE
Race condition for binding of properties for backend selectors

### ISSUE THREAD

**Issue by djkunkel** [Sitefinity/feather-widgets#2149]
We've noticed on projects occasionally when opening a backend editor, selectors are at default values even if values are present for them.

When implementing this tutorial to add a custom taxonomy selector:
http://docs.sitefinity.com/feather-hierarchical-taxon-selector
I noticed that sometimes my editor would be set to Categories which is the default for sf-taxonomy-selector. I believe that occasionally binding finishes before the data for properties is loaded, and thus the defaults. On other built-in widgets this manifests as an empty selector which can cause users to erase their previous selections accidentally.

I worked around using al-if to guard binding of my widget view until data is loaded.  Because al-if creates a child scope, I had to create a $scope.model object to store the values I wanted to access from inside the al-if'ed elements that needed special processing (JSON parsing in my example).

**Comment by CodeAnimal** [comment]
Yes I have this issue, and it happens very regularly. I'd say this is a pretty significant bug, that causes users (especially those that aren't too technical) issues almost daily.
